v1.1/known-limitations.md

```diff
@@ -74,6 +74,12 @@ Also, a prepared [`INSERT`](insert.html), [`UPSERT`](upsert.html), or [`DELETE`]
 - If the number of columns has increased, the prepared statement returns an error but nonetheless writes the data.
 - If the number of columns remains the same but the types have changed, the prepared statement writes the data and does not return an error.
 
+## Join flags when restarting a cluster with different addresses
+
+In all our deployment tutorials, we provide the addresses of the first few nodes in the cluster to the `--join` flag when starting each node. In a new cluster, this ensures that all nodes are able to learn the location of the first key-vlaue range, which is part of a meta-index identifying where all range replicas are stored, and which nodes require to initialize themselves and start accepting incoming connections. Each node also persists the addresses of all other nodes in the cluster on disk such that it can reconnect to them if the nodes in the `--join` flag ever happen to be unavailable when restarting. This ensures that a restarting node will always be able to connect to a node with a copy of the first range even if they're no longer located on the nodes in the `--join` flag.
+
+However, if the nodes in a cluster are restarted with different addresses for some reason, then it's not guaranteed that a copy of the first range will be on the nodes in the join flags. In such cases, the `--join` flags must form a fully-connected directed graph. The easiest way to do this is to put all of the new nodes' addresses into each node's `--join` flag, which ensures all nodes can join a node with a copy of the first key-value range.
```

Comment:
A fully-connected graph is not strictly required (connecting each node to nodes 1..N-1 leaves the last node with no inbound edges). I'd just drop that sentence and leave the recommendation to put all the addresses in all the join flags.

Reply:
If you've reduced the replication factor of the first range to 1 and its only copy is on the last node, then that configuration won't work. I do think it has to be fully-connected to guarantee success in cases like that. I can remove the wording, but it's also how we documented it in the 1.0 docs. Should I remove it there too?

Reply:
I'm fine leaving it as-is.